### PR TITLE
[FX-5570] Fix visual problems when as Link

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-5570]

### Description

When adopting the latest version in the Staff Portal, I found few visual issues with `<Button as={Link} />`

### How to test

- [Temploy](https://picasso.toptal.net/fx-null-fix-button-2)
- FIXME: Add the steps describing how to verify your changes

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- Annotate all `props` in component with documentation
- Create `examples` for component
- Ensure that deployed demo has expected results and good examples
- Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5570]: https://toptal-core.atlassian.net/browse/FX-5570?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: packages/base/Button/src/ButtonBase/ButtonBase.tsx [hunk 1]
```diff
@@ -103,6 +103,8 @@ export const ButtonBase: OverridableComponent<Props> = forwardRef<
     rootElementName ||
     (as !== 'button' && ('href' in props || 'to' in props) ? 'a' : undefined)
 
+  console.log(finalRootElementName)
```
Comment (augustobmoura) [inline packages/base/Button/src/ButtonBase/ButtonBase.tsx:106]: This is probably a leftover

**Review by augustobmoura — COMMENTED** [review]
What was the bug? I can't reproduce any issue with underline on master's storybook

**Comment by TomasSlama** [comment]
The bug was when you use `<Button as={PicassoLink} />` when you hover, it is underlined, when you pass disabled to secondary button, it has blue text color

File: packages/base/Button/src/ButtonBase/ButtonBase.tsx (hunk 1 above)
Comment (TomasSlama) [reply packages/base/Button/src/ButtonBase/ButtonBase.tsx]: removed

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: packages/base/Button/src/Button/Button.tsx [hunk 2]
```diff
@@ -68,6 +68,8 @@ export interface Props
   title?: string
   /** HTML type of Button component */
   type?: 'button' | 'reset' | 'submit'
+  /** The HTML element that is ultimately rendered, for example 'button', 'a' or 'label */
+  rootElementName?: keyof HTMLElementTagNameMap
```
Comment (TomasSlama) [inline packages/base/Button/src/Button/Button.tsx:72]: It is needed for some edge cases in Client portal, where our automatic detection is insufficient.

Comment (augustobmoura) [inline packages/base/Button/src/Button/Button.tsx:72]: Is this prop being used anywhere, or is it just being passed down by `...rest`?

Comment (augustobmoura) [reply packages/base/Button/src/Button/Button.tsx:72]: Do we need to provide this prop for people to use? Maybe we can skip it and have only `as`?

Comment (ruslan-sed) [inline packages/base/Button/src/Button/Button.tsx:72]: How is it used here?

Comment (TomasSlama) [reply packages/base/Button/src/Button/Button.tsx:72]: https://toptal-core.slack.com/archives/CERF5NHT3/p1719253939042919

**Review by ruslan-sed — APPROVED** [review]
Works as expected. But there is one unclear thing about `rootElementName` which looks as unused prop.

File: packages/base/Button/src/Button/Button.tsx (hunk 2 above)
Comment (augustobmoura) [reply packages/base/Button/src/Button/Button.tsx:72]: I read the thread, but I still think we shouldn't allow this escape hatch. We are instead introducing more edge cases and shifting responsibility to users to workaround `as`. If this is not strictly necessary I would avoid [introducing it](https://en.wikipedia.org/wiki/Law_of_Demeter)

Comment (TomasSlama) [reply packages/base/Button/src/Button/Button.tsx:72]: It is just passed to the ButtonBase

**Review by augustobmoura — APPROVED** [review]

**Pull request merged**